### PR TITLE
return failure details when unauthorized to push

### LINKED ISSUE #1140: Model push is not working
I tried many times to push my models to ollama but always I get this error
```
retrieving manifest
Error: max retries exceeded
```

### PR DESCRIPTION
**Pull request by BruceMacD**
Previous behavior:
Pushing to the default namespace or a namespace you don't have access to results in a vague error.
```
$ ollama push mario
retrieving manifest
Error: max retries exceeded
```

New behavior:
Pushing to the default namespace or a namespace you don't have access to results the reason for the error.
```
$ ollama push mario
retrieving manifest
Error: unable to push library/mario, make sure this namespace exists and you are authorized to push to it

$ ollama push bruxe/mario
retrieving manifest
Error: unable to push bruxe/mario, make sure this namespace exists and you are authorized to push to it
```

Resolves #1140

### CODE REVIEW

File: server/images.go [hunk 1]
```diff
@@ -1187,7 +1192,10 @@ func makeRequestWithRetry(ctx context.Context, method string, requestURL *url.UR
 				body.Seek(0, io.SeekStart)
 			}
 
-			continue
+			if try == (maxRetries - 1) {
```
Comment (jmorganca) [inline server/images.go:1195]: could we set an outer `err` variable instead and then just return it at the end? feels strange to check boundary conditions

Comment (mxyng) [reply server/images.go:1195]: I agree. It's cleaner to set a `lastError` and return. Other similar functions follow that pattern already.

```go
var lastError error
for try := 0; try < maxRetries; try++ {
  ...
  switch resp.StatusCode {
  case http.StatusUnauthorized:
    lastError = fmt.Errorf("unauthorized: make sure you have access to this model")
    continue
  }
}

return nil, fmt.Errorf("%w: %w", errMaxRetriesExceeded, lastError)
```

This also enables the function to handle retries for other status codes

Comment (BruceMacD) [reply server/images.go]: Made this change 👍

File: server/images.go [hunk 2]
```diff
@@ -1165,7 +1168,10 @@ func GetSHA256Digest(r io.Reader) (string, int64) {
 	return fmt.Sprintf("sha256:%x", h.Sum(nil)), n
 }
 
+var errUnauthorized = fmt.Errorf("unauthorized")
+
 func makeRequestWithRetry(ctx context.Context, method string, requestURL *url.URL, headers http.Header, body io.ReadSeeker, regOpts *RegistryOptions) (*http.Response, error) {
+	var lastErr error
```
Comment (jmorganca) [inline server/images.go:1174]:
```suggestion
	lastErr:= errMaxRetriesExceeded
```

**Review by jmorganca — APPROVED** [review]

**Pull request merged**